junebug/tests/test_utils.py

```diff
@@ -298,3 +299,99 @@ def test_api_from_status(self):
             'message': 'Bar',
             'details': {'baz': 'quux'}
         })
+
+    def test_public_http_properties_explicit(self):
+        result = channel_public_http_properties({
+            'web_path': '/baz/quux',
+            'web_port': 2121,
+            'public_http': {
+                'web_path': '/foo/bar',
+                'web_port': 2323,
+            }
```

Comment:
Would be nice to be able to automatically assign ports or to use named Unix sockets instead of TCP? Maybe something for a different PR.